### ISSUE TITLE
How to put text of “Questionnaire-ID” into tex?

### ISSUE THREAD

**Issue by MarDos1234** [sdaps/sdaps#237]
Hi all,
I am a new in tex and sdaps too.
I can do questionnaire if specific student's ID (8 digits) which is coded into QR code, but I would like to automatically write this ID into some place into questionnaire (for example right corner) so student can easily find his/her test before filing it and I can eliminate recognition error in ID or name.
Exist some variable name for questionnaire ID or something like this?

Thank you for help.

**Comment by benzea** [comment]
You can use `\qid`.

**Comment by benzea** [comment]
I should have said … be really careful that the text will not re-flow around it causing a relayout of the document.

As such, I recommend using `\strut` and making sure there is enough space (i.e. forced newline). Or some other method.

**Comment by MarDos1234** [comment]
Thanks benzea, but I stilldonot have wanted result.
I use code:
	\begin{info}
	Your ID: \qid
	\end{info}

And when pdflatex create pdf, it seems OK:
![image](https://user-images.githubusercontent.com/57493520/143435933-796251d8-1b90-4cf1-bb4a-3f4de0670b55.png)

When I setup new project, it seems good too:
![image](https://user-images.githubusercontent.com/57493520/143436177-a4794f14-5f7f-4da6-bf60-e89745e7adad.png)

But when I call sdaps stamp -f ids.txt MyProject error message occure:
![image](https://user-images.githubusercontent.com/57493520/143436280-81fa4fed-18d1-46fd-aa67-af74c4c51711.png)

When I erase \qid part stamp works well.

Can you help me please?

**Comment by benzea** [comment]
Hrm, I should hide the backtrace. There is a message `Temporary files left in '/tmp/sdaps-stamp-4kqvw5gh'` in this case. Look at the directory and have a look a the `.log` file.

Are the IDs purely numeric?

**Comment by MarDos1234** [comment]
No IDs are not purely numeric. When I use numeric IDs it works. So I cannot use letters in ID when I want to use \qid. Good to know.

Thank you for your help.

**Comment by MarDos1234** [comment]
As I wrote before, when I used non numeric ID plus \qid and stamp.
I am sending log file from Temporary files left in '/tmp/sdaps-stamp-4kqvw5gh'

[questionnaire.log](https://github.com/sdaps/sdaps/files/7606160/questionnaire.log)

**Comment by benzea** [comment]
Yeah, but I don't think it is just non-numeric. I actually have a test with some special characters, so unless it broke recently I should have noticed.

Maybe spaces do it, or something else entirely.

**Comment by benzea** [comment]
Ahh, missing $ inserted. So you have a character that triggers math mode. Like `_`.

**Comment by benzea** [comment]
OK, so this is quite obvious. We are just passing over the string right now.

I suppose, running it through our `raw_unicode_to_latex` helper instead of just replacing braces would make sense. Though not sure what that means for the QR code/code-128 generators. The solution might be to pass over two string from the start, one for the barcode generator and one for displaying.
